src/hmem_cuda.c

```diff
@@ -171,19 +181,52 @@ int cuda_hmem_init(void)
 		goto err_dlclose_cuda;
 	}
 
+	cuda_ops.cuInit = dlsym(cuda_handle, "cuInit");
+	if (!cuda_ops.cuInit) {
+		FI_WARN(&core_prov, FI_LOG_CORE, "Failed to find cuInit\n");
+		goto err_dlclose_cuda;
+	}
+
 	return FI_SUCCESS;
 
 err_dlclose_cuda:
 	dlclose(cuda_handle);
 err_dlclose_cudart:
```

Comment:
Is this label used?

Reply:
Yep. If dlopen of libcuda.so fails, it jumps here.

Reply:
I didn't expand the code enough to see that.  Thanks.  Waiting on CI, but I think we want this in today, so it hits rc2.